inference-engine/src/mkldnn_plugin/nodes/mkldnn_roll_node.cpp

```diff
@@ -0,0 +1,214 @@
+    if (axesTensorPrec != Precision::I32 && axesTensorPrec != Precision::I64) {
+        IE_THROW() << layerErrorPrefix << " has unsupported 'axes' input precision: " << axesTensorPrec.name();
+    }
+
+    const auto axesTensorRank = axesTensor.getDims().size();
+    if (axesTensorRank > 1) {
+        IE_THROW() << layerErrorPrefix << " doesn't support 'axes' input tensor with rank: " << axesTensorRank;
+    }
+
+    /* Shift */
+    auto shiftData = layer->insData[SHIFT_INDEX].lock();
+    if (shiftData == nullptr) {
+        IE_THROW() << layerErrorPrefix << " has nullable 'shift' data";
+    }
+    const auto& shiftTensor = shiftData->getTensorDesc();
+    const auto& shiftTensorPrec = shiftData->getTensorDesc().getPrecision();
+    if (shiftTensorPrec != Precision::I32 && shiftTensorPrec != Precision::I64) {
+        IE_THROW() << layerErrorPrefix << " has unsupported 'shift' input precision: " << shiftTensorPrec.name();
+    }
+
+    const auto shiftTensorRank = shiftTensor.getDims().size();
+    if (shiftTensorRank > 1) {
+        IE_THROW() << layerErrorPrefix << " doesn't support 'shift' input tensor with rank: " << shiftTensorRank;
+    }
+
+    shape = dataShape;
+}
+void MKLDNNRollNode::getSupportedDescriptors() {}
+
+void MKLDNNRollNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto inputData = getCnnLayer()->insData[0].lock();
+
+    if (inputData == nullptr) {
+        IE_THROW() << "Roll layer with name '" + getCnnLayer()->name + "'" << " has nullable data";
+    }
+
+    InferenceEngine::Precision precision = inputData->getPrecision();
+
+    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+
+    auto srcDims = getParentEdgeAt(0)->getDims();
+
+    auto dataMemoryFormat = MKLDNNMemory::GetPlainFormat(getParentEdgeAt(0)->getDims());
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = true;
+
+    auto createDataConfig = [](const MKLDNNDims& dims, memory::data_type dataType) -> InferenceEngine::DataConfig {
+        InferenceEngine::DataConfig dataConfig;
+        dataConfig.inPlace = -1;
+        dataConfig.constant = false;
+        dataConfig.desc = MKLDNNMemoryDesc(dims, dataType, MKLDNNMemory::GetPlainFormat(dims));
+        return dataConfig;
+    };
+
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(0)->getDims(), dataType));
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(1)->getDims(), memory::data_type::s32));
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(2)->getDims(), memory::data_type::s32));
+
+    config.outConfs.push_back(createDataConfig(getChildEdgeAt(0)->getDims(), dataType));
+
+    supportedPrimitiveDescriptors.push_back({config, impl_desc_type::ref, dataMemoryFormat});
+}
+
+
+void MKLDNNRollNode::execute(mkldnn::stream strm) {
+    auto input = getParentEdgeAt(DATA_INDEX)->getBlob();
```

Comment:
Please avoid using getBlob method since it might be quite slow. The proper way to get pointers: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp#L315.
BTW you can get the pointers right inside `rollImpl` so not necessary to pass them as parameters.